casanovo/data/ms_io.py

```diff
@@ -2,18 +2,56 @@
 
 import collections
 import csv
+import dataclasses
 import operator
 import os
 import re
 from pathlib import Path
-from typing import List
+from typing import List, Tuple
 
 import natsort
 
 from .. import __version__
 from ..config import Config
 
 
+@dataclasses.dataclass
+class PepSpecMatch:
+    """
+    Peptide Spectrum Match (PSM) dataclass
+
+    Parameters
+    ----------
+    sequence : str
+        The amino acid sequence of the peptide.
+    spectrum_id : Tuple[str, str]
+        A tuple containing the spectrum identifier in the form
+        (spectrum file name, spectrum file idx)
+    peptide_score : float
+        Confidence score of the match between the full peptide sequence and the
+        spectrum.
+    charge : int
+        The charge state of the peptide ion observed in the spectrum.
+    expected_mz : float
+        The expected mass-to-charge ratio (m/z) of the peptide based on its
+        sequence and charge state.
+    actual_mz : float
+        The observed mass-to-charge ratio (m/z) of the peptide as detected in
+        the spectrum.
+    aa_scores : List[float]
```

Comment:
**todo:** I think this is actually a NumPy array rather than a list coming from the model. You could modify it to `Iterable`, that fits both types and covers the downstream functionality.